lib/spack/spack/test/concretization/core.py

```diff
@@ -759,19 +759,22 @@ def test_virtual_is_fully_expanded_for_mpileaks(self):
     @pytest.mark.parametrize(
         "spec_str,expected,not_expected",
         [
+            # TODO/TBD: This doesn't appear to be the case anymore with llvm, which
+            # TODO/TBD: is what gets picked now for mpileaks.
+            # TODO/TBD: Should mpileaks be dropped or replaced?
             # clang only provides C, and C++ compilers, while gcc has also fortran
             #
             # If we ask mpileaks%clang, then %gcc must be used for fortran, and since
             # %gcc is preferred to clang in config, it will be used for most nodes
             (
                 "mpileaks %clang",
-                {"mpileaks": "%clang", "libdwarf": "%gcc", "libelf": "%gcc"},
-                {"libdwarf": "%clang", "libelf": "%clang"},
+                {"mpileaks": "%clang", "libdwarf": "%clang", "libelf": "%clang"},
+                {"libdwarf": "%gcc", "libelf": "%gcc"},
             ),
```

Comment:
@alalazo Under the circumstances and given the other test parameters, should these `mpileaks` tests be dropped? Is there a reasonable or preferred alternative from the mock packages?

Reply:
Ugh, this seems some sort of cross-talking between tests, or something that is order dependent. I'll check why this is happening.

Reply:
Very likely, this happened because we made `llvm` provide `fortran` too :melting_face: I'll try to think of another way to test this kind of scenario.